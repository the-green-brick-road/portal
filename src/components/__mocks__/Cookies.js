/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }             from '@mui/material';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color } = props;

    /* -------- Defining theme --------- */
    return (
        <Box style={{ minHeight:'30px',zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> Cookies </Typography>
        </Box>
    );

}

export default Cookies;
