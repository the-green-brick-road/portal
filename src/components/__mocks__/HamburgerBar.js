/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerBar component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height, isNegative, theme={} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }

    return (
        <Box style={{height:height, backgroundColor:color, width: '100%'}}>
            <Typography> HamburgerBar </Typography>
        </Box>
    );

}

export default HamburgerBar;
