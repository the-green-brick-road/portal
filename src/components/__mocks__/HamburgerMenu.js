/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerMenu component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerMenu(props) {

    /* --------- Gather inputs --------- */
    const { top , margin, isNegative, theme={} } = props;
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }
    const menuWidthString = `calc( 100vw - 2 * ${margin})`;

    return (
        <Box style={{ minHeight:'30px', width:menuWidthString, top:top, zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> HamburgerMenu </Typography>
        </Box>
    );

}

export default HamburgerMenu;