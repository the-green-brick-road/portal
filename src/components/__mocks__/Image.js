/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'blue' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Image;
