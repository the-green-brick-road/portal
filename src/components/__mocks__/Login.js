/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Login component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @29 may 2023
# Latest revision: 29 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Login(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'green' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Login;
