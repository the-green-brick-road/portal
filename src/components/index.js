/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Components exports
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @18 may 2023
# Latest revision: 16 june 2023
# ---------------------------------------------------- */

export { default as NavigationBar } from './navigationbar/NavigationBar';
export { default as HamburgerBar }  from './hamburgermenu/HamburgerBar';
export { default as HamburgerMenu } from './hamburgermenu/HamburgerMenu';
export { default as Image }         from './image/Image';
export { default as Footer }        from './footer/Footer';
export { default as Cookies }       from './cookies/Cookies';