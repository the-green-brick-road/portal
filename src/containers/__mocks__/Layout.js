/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* React includes */
import { Fragment }   from 'react';
import { Outlet }     from "react-router-dom";

/* Material UI includes */
import { Typography } from '@mui/material';

function Layout(props) {

    return (

        <Fragment>
            <Typography> Mock Layout </Typography>
            <Outlet />
        </Fragment>

    );

}

export default Layout;
