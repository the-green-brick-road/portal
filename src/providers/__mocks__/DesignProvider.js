/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Design provider mock
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @09 may 2023
# Latest revision: 09 may 2023
# ---------------------------------------------------- */

/* React includes */
import { createContext }       from 'react';

/* Material UI includes */
import { CssBaseline }                                 from '@mui/material';
import { ThemeProvider, createTheme }                  from '@mui/material/styles';

const context = createContext(null);

/* ----- Define provider values ---- */
var state = {
    isWebpSupported: true,
    isSliding:       false,
    isDarkMode:      false,
    images:          {},
    screen:          '',
    sizes:           {},
    setIsSliding:    jest.fn((value) => { state.isSliding = value }),
};

function DesignProvider(props) {

    const { children, isWebpSupported, screen, isDarkMode = false, isSliding = false, sizes, images = [], theme = {} } = props;

    /* ------ Defining behaviour ------- */
    var local_theme = {}
    if (Object.keys(theme).length !== 0) { local_theme = createTheme(theme); }

    state.isWebpSupported = isWebpSupported;
    state.screen          = screen;
    state.images          = {}
    state.sizes           = sizes;
    state.isSliding       = isSliding;
    state.isDarkMode      = isDarkMode;
    for (const image of images)
    {

        const rawext = image["raw"].split('.').pop();
        const webext = image["web"].split('.').pop();

        state.images[image.name] = {}
        state.images[image.name][rawext] = {}
        state.images[image.name][webext] = {}

        state.images[image.name][rawext]['small']           = {}
        state.images[image.name][rawext]['small']['img']    = require(`../../assets/small/${image.raw}`)
        state.images[image.name][rawext]['small']['width']  = `${sizes['small-width']}w`
        state.images[image.name][rawext]['medium']          = {}
        state.images[image.name][rawext]['medium']['img']   = require(`../../assets/medium/${image.raw}`)
        state.images[image.name][rawext]['medium']['width'] = `${sizes['medium-width']}w`
        state.images[image.name][rawext]['large']           = {}
        state.images[image.name][rawext]['large']['img']    = require(`../../assets/${image.raw}`)
        state.images[image.name][rawext]['large']['width']  = `${sizes['large-width']}w`

        state.images[image.name][webext]['small']           = {}
        state.images[image.name][webext]['small']['img']    = require(`../../assets/small/${image.web}`)
        state.images[image.name][webext]['small']['width']  = `${sizes['small-width']}w`
        state.images[image.name][webext]['medium']          = {}
        state.images[image.name][webext]['medium']['img']   = require(`../../assets/medium/${image.web}`)
        state.images[image.name][webext]['medium']['width'] = `${sizes['medium-width']}w`
        state.images[image.name][webext]['large']           = {}
        state.images[image.name][webext]['large']['img']    = require(`../../assets/${image.web}`)
        state.images[image.name][webext]['large']['width']  = `${sizes['large-width']}w`

        state.images[image.name]['default']= require(`../../assets/${image.raw}`)

    }

    /* ----------- Define HTML --------- */
    return (
        <ThemeProvider theme={local_theme}>
            <CssBaseline />
            <context.Provider value={state}>
                {children}
            </context.Provider>
        </ThemeProvider>
    );

}

function useDesign() { return state; };

export { useDesign, DesignProvider };