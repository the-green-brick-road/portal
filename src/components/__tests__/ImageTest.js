/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Analytics Provider test suite
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Test includes */
import { render, screen, prettyDOM } from '@testing-library/react'
import { expect, test }              from '@jest/globals';

/* Component under test */
import { Image }                     from '../../components';

/* Mocks includes */
/* eslint-disable jest/no-mocks-import */
import { useDesign as mockUseDesign, DesignProvider as MockDesignProvider } from '../../providers/__mocks__/DesignProvider';
jest.mock('../../providers', () => ({ useDesign: (() => { return mockUseDesign(); }) }));
/* eslint-enable jest/no-mocks-import */


describe("Image component" ,() => {

    test('Should display an image if existing', async () => {

        const images = [
            {
                "name" : "test",
                "raw"  : "test.png",
                "web"  : "test.webp",
            }
        ]

        const sizes = {
            "small-width"          : 200,
            "medium-width"         : 400,
            "large-width"          : 800,
        }

        const view = render(

            <div>
                <MockDesignProvider screen='large' images={images} sizes={sizes}>
                    <Image name='test' style={{ margin:'70px' }}/>
                </MockDesignProvider>
            </div>

        );

        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true,highlight: false});
        expect(tree).toMatchSnapshot();

    })

    test('Should not display an image if not existing', async () => {

        const images = [
            {
                "name" : "test",
                "raw"  : "test.png",
                "web"  : "test.webp",
            }
        ]

        const sizes = {
            "small-width"          : 200,
            "medium-width"         : 400,
            "large-width"          : 800,
        }


        render(

            <div>
                <MockDesignProvider screen='large' images={images} sizes={sizes}>
                    <Image name='logo' style={{ margin:'70px' }}/>
                </MockDesignProvider>
            </div>

        );
        expect(screen.queryByTestId('picture-test')).toBeNull();

    })

})
