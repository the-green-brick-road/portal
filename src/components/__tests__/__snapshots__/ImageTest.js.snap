// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component Should display an image if existing 1`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <picture
        data-testid=\\"picture-test\\"
        style=\\"margin: 70px;\\"
      >
        <source
          srcset=\\" test.webp 200w , test.webp 400w , test.webp 800w ,\\"
          type=\\"image/webp\\"
        />
        <source
          srcset=\\" test.png 200w , test.png 400w , test.png 800w ,\\"
          type=\\"image/png\\"
        />
        <img
          alt=\\"test\\"
          src=\\"test.png\\"
          style=\\"margin: 70px;\\"
        />
      </picture>
    </div>
  </div>
</body>"
`;
